engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/extractor/DataSourceSqlExtractor.scala

```diff
@@ -0,0 +1,127 @@
+package pl.touk.nussknacker.engine.flink.table.extractor
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.flink.table.api.{EnvironmentSettings, TableEnvironment, TableResult}
+import org.apache.flink.table.catalog.{CatalogBaseTable, ObjectPath}
+import org.apache.flink.table.types.DataType
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.flink.table.extractor.SqlDataSourceConfig.Connector
+import pl.touk.nussknacker.engine.flink.table.extractor.SqlStatementReader.SqlStatement
+import pl.touk.nussknacker.engine.flink.table.extractor.TypeExtractor.extractTypedSchema
+
+import scala.util.Try
+
+// TODO: additional validation (maybe outside scope of this class' responsibility):
+//  - duplicate table names
+//  - custom validation per connector
+object DataSourceSqlExtractor extends LazyLogging {
+
+  import scala.jdk.CollectionConverters._
+  import scala.jdk.OptionConverters.RichOptional
+
+  private val connectorKey        = "connector"
+  private val builtInCatalogName  = "defaultCatalog"
+  private val buildInDatabaseName = "defaultDatabase"
+
+  def extractTablesFromFlinkRuntime(
+      createTableStatements: List[SqlStatement]
+  ): List[Either[SqlExtractorError, SqlDataSourceConfig]] = {
+    val settings = EnvironmentSettings
+      .newInstance()
+      .withBuiltInCatalogName(builtInCatalogName)
+      .withBuiltInDatabaseName(buildInDatabaseName)
+      .build()
+    val tableEnv = TableEnvironment.create(settings)
```

Comment:
I see some a problem here. This method creates fat objects like `ResourceManager`. Each of them are `Closable` but `TableEnvironment` isn't. I see that it is on the Flink's roadmap in https://cwiki.apache.org/confluence/display/FLINK/FLIP-36%3A+Support+Interactive+Programming+in+Flink. This method will be run multiple times, e.g. we have a model reloading mechanism :-/ Do you have some idea how to deal with this?

Reply:
Let's leave a TODO

Reply:
Added a TODO with a link to this comment